src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+					notifySysAdminsAboutInActiveIsFrozenSysAdminUser();
+					breakOuterCircle = true;
+					break;
+				}
+			}
+			if(breakOuterCircle) {
+				break;	
+			}
+		}
+	}
+
+
+	/*********************************************************************************************************
+    * @description Logs an Error__c record and emails the org's System Administrators.
+    * @return void
+    **********************************************************************************************************/
+	@testVisible
+	private static void notifySysAdminsAboutInActiveIsFrozenSysAdminUser() {
+		logInActiveIsFrozenUserError();
+		sendInActiveIsFrozenUserEmail();
+	}
+
+
+	/*********************************************************************************************************
+    * @description Logs an Error__c record.
+    * @return void
+    **********************************************************************************************************/
+	@testVisible
+	private static void logInActiveIsFrozenUserError() {
+		Error__c inActiveIsFrozenError = new Error__c(Error_Type__c = ERROR_TYPE_SCHEDULED_APEX,
+														Object_Type__c = OBJECT_TYPE_USER,
+														Datetime__c = System.now(), 
+														Full_Message__c = ERROR_MESSAGE_SCHEDULED_APEX);
+
+		Database.insert(inActiveIsFrozenError, true);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Emails the org's System Administrators.
+    * @return void
+    **********************************************************************************************************/
+	@testVisible
+	private static void sendInActiveIsFrozenUserEmail() {
+		ERR_Notifier.sendErrorNotifications(CONTEXT_TYPE_USER);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Utility method to retrieve the System Adminstrator's ProfileId.
+    * @return Id The Id of the System Administrator Profile.
+    **********************************************************************************************************/
+	private static Id retrieveSysAdminProfileId() {
+		Id sysAdminProfileId = null; 
+
+		List<Profile> sysAdminProfile = [SELECT Id FROM Profile WHERE Name = :SYSTEM_ADMINISTRATOR];
+		if(!sysAdminProfile.isEmpty()) {
+			sysAdminProfileId = sysAdminProfile[0].Id;
+		}
+		
+		return sysAdminProfileId;
+	}
+	
+
+	/*********************************************************************************************************
+    * @description Retrieves the UserLogin records for Users.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsFrozen status.
+    **********************************************************************************************************/
+	private static Map<Id, Boolean> retrieveUserLogins(List<User> updatedUsers) {
```

Comment:
this method seems misnamed.  it sounds like it will be returning UserLogin records, but in fact it is returning the map of whether each user is frozen or not.  Might be better named retrieveUserFrozenStatus, or something like that.  the fact that the frozen status is derived from the UserLogin record is not really the concern of the caller.

Reply:
Updated this method to reflect its original intention. 

Good catch!